src/views/requestAccess.js

```diff
@@ -1,146 +1,123 @@
-				  	<div className="dropdown-content dropdown-content-width">
-					    <DropDownItem dropContent="JOIN Staff" clickHandle={handleClickJOIN} />
-					    <DropDownItem dropContent="Property Manager" clickHandle={handleClickProperty} />
-				  	</div>
-			    </div>
-			</div>
-			<div>
-				{viewPropertyManagerOptions && <PropertyManagerOptions />}
-				{viewJoinStaffOptions && <JoinStaffOptions />}
-			</div>
-		</div>
-	);
+const RoleDropDown = (props) => {
+  console.log(props.selectionOptions);
+  console.log(typeof props.selectionOptions);
+  let roles = props.selectionOptions;
+  return (
+    <div className="section-row">
+      <div className="select is-rounded">
+        <select
+          defaultValue='default'
+          onChange={e => props.selectionHandler(e.target.value)}>
+          <option value='default' disabled defaultValue>Select Role</option>
+          {
+            props.selectionOptions.map((role) => 
+            <option>{role}</option>
+            )
+          }
+        </select>
+      </div>
+    </div>
+  )
 }
 
 // Section under CONTACT
-export const InfoField = ({label, info}) => {
+export const InfoField = ({label, info, changeHandler}) => {
 	const infoField = "has-text-weight-bold " + ((label === "Phone") ? "phone-field" : (label === "Email" ? "email-field" : "name-field"));
 	return (
 		<div>
 			<hr className="line" ></hr>
 			<span className="input-field"> {label}
-				<p className={infoField} contentEditable>
-			        {info}
-			    </p>
+				<input className={`contact-input ${infoField}`} type="text" defaultValue={info} onChange={(e) => changeHandler(e.target.value)}>
+			    </input>
 		    </span>
 	    </div>
 	);
 }
 
 export const RequestAccess = (props) => {
-	// const id = props.match.params.id;
+  // Get context for API auth header
+  const userContext = useContext(UserContext);
+
+  const [roleObject, setRoleObject] = useState({});
+  const [selectionOptions, setSelectionOptions] = useState([]);
+  const [currentSelection, selectionHandler] = useState("");
+  const [fName, setFirstName] = useState("");
+  const [lName, setLastName] = useState("");
+  const [emailAddress, setEmail] = useState("");
+
+  useEffect(() => {
+    console.log(process.env.REACT_APP_API_URL)
+    axios.get(`/api/roles`)
+    .then((response) => {
+      let data = JSON.parse(response.data);
+      // Get key value object of all roles
+      setRoleObject(data);
+      // Get Role names
+      let roleArray = Object.keys(data);
+      // Removes "Pending" role
+      roleArray.shift();
+      // Get array of roles to map to selection options
+      setSelectionOptions(roleArray);
+    })
+    .catch((error) => {
+      alert(error);
+      console.log(error);
+    });
+  }, []);
+
+  const grantAccess = (role, firstName, lastName, email, id) => {
+    // Get role index from roleObject
+    let roleID = roleObject[role];
+    axios.patch(`/api/user/${id}`, {
```

Comment:
Since I didn't get any roles, the dropdown still said "Select Role", and clicking the Grant Access button gives me a 400 Bad Request since a role ID wasn't sent along with the request.
![image](https://user-images.githubusercontent.com/8600792/90592124-b28c3380-e199-11ea-9e8b-4a165dbb56ae.png)

Let's put in a condition on the Grant Access button that disables it until a role is selected